gravitee-apim-e2e/api-test/use-case-test/src/management/publisher/properties/add-properties-and-use-them.spec.ts

```diff
@@ -13,8 +13,134 @@
+import { PropertyEntity } from '@model/apis';
+import { fetchGatewaySuccess } from '@gravitee/utils/gateway';
+
+const orgId = 'DEFAULT';
+const envId = 'DEFAULT';
+const apisManagementApiAsApiUser = new APIsApi(forManagementAsApiUser());
 
 describe('Add properties and use them', () => {
-  test.skip('To complete', async () => {});
+  let createdApi: ApiEntity;
+
+  beforeAll(async () => {
+    // create APIs with a published keyless plan and mock policy
+    createdApi = await apisManagementApiAsApiUser.importApiDefinition({
+      envId,
+      orgId,
+      body: ApisFaker.apiImport({
+        plans: [
+          PlansFaker.plan({
+            security: PlanSecurityType.KEY_LESS,
+            status: PlanStatus.PUBLISHED,
+            flows: [
+              {
+                name: '',
+                path_operator: {
+                  path: '/',
+                  operator: PathOperatorOperatorEnum.STARTS_WITH,
+                },
+                condition: '',
+                consumers: [],
+                methods: [],
+                pre: [
+                  {
+                    name: 'Mock',
+                    description: '',
+                    enabled: true,
+                    policy: 'mock',
+                    configuration: {
+                      content: '{ "key_1": "{#properties[\'key_1\']}", "encryptedKey": "{#properties[\'encryptedKey\']}" }',
+                      status: '200',
+                    },
+                  },
+                ],
+                post: [],
+                enabled: true,
+              },
+            ],
+          }),
+        ],
+      }),
+    });
+
+    // start it
+    await apisManagementApiAsApiUser.doApiLifecycleAction({
+      envId,
+      orgId,
+      api: createdApi.id,
+      action: LifecycleAction.START,
+    });
+  });
+
+  test('should set API property', async () => {
+    const newProperties: PropertyEntity[] = [
+      {
+        key: 'key_1',
+        value: 'abc',
+        encrypted: false,
+        dynamic: false,
+        encryptable: false,
+      },
+      {
+        key: 'encryptedKey',
+        value: 'secretKeyValue',
+        encrypted: true,
+        dynamic: false,
+        encryptable: false,
+      },
+    ];
+
+    const updateApiEntity = UpdateApiEntityFromJSON({
+      ...createdApi,
+      properties: newProperties,
+    });
+
+    const updatedApi = await succeed(
+      apisManagementApiAsApiUser.updateApiRaw({
+        api: createdApi.id,
+        updateApiEntity,
+        orgId,
+        envId,
+      }),
+    );
+
+    await succeed(apisManagementApiAsApiUser.deployApiRaw({ orgId, envId, api: updatedApi.id }));
+
+    expect(updatedApi.properties.length).toEqual(2);
+    expect(updatedApi.properties[0].key).toEqual('key_1');
+    expect(updatedApi.properties[0].value).toEqual('abc');
+    expect(updatedApi.properties[1].key).toEqual('encryptedKey');
+    expect(updatedApi.properties[1].value).toEqual('secretKeyValue');
```

Comment:
Could be easier to read if everything is in a single expect:

```suggestion
    expect(updatedApi.properties).toEqual([
      { key: 'key_1', value: 'abc' },
      { key: 'encryptedKey', value: 'secretKeyValue' },
    ]);
```

Reply:
Yes, that's true, thanks for your feedback! 👍🏻